modules/c/pages/working-with-vector-search.adoc

```diff
@@ -0,0 +1,230 @@
+= Working with Vector Search
+:page-status: Beta
+:page-edition: Enterprise
+:page-aliases: 
+ifdef::show_edition[:page-edition: {release}]
+ifdef::prerelease[:page-status: {prerelease}]
+:page-role:
+:description: Use Vector Search with Full Text Search and Query.
+:keywords: edge AI api swift ios macos apple vector search generative
+
+[abstract]
+{description}
+
+== Use Vector Search
+
+To configure a project to use vector search, follow the xref:c:gs-install.adoc[installation instructions] to add the Vector Search extension.
+
+NOTE: You must install Couchbase Lite to use the Vector Search extension.
+
+== Create a Vector Index
+
+This method shows how you can create a vector index using the Couchbase Lite Vector Search extension.
+
+[source, c]
+----
+
+include::c:example$code_snippets/VectorSearch.h[tags=vs-create-default-config]
+
+----
+
+First, we initialize the `config` object with a new `VectorIndexConfiguration` object initialized with the following parameters:
+
+* The expression of the data as a vector.
+
+* The width or `dimensions` of the vector index is set to `3`.
+
+* The amount of `centroids` is set to `2`.
+This means that there will be two buckets with a single centroid each that gathers together similar vectors.
+
+There are also some default values that we did not set with the `VectorIndexConfiguration()` method, these are:
+
+* The encoding type - 8 bit Scalar Quantization.
+
+* The distance metric - Squared Euclidean.
+
+* The minimum and maximum number of vectors for training the index.
+
+The method below shows the generation of a custom vector index configuration.
+
+[source, c]
+----
+
+include::c:example$code_snippets/VectorSearch.h[tags=vs-create-custom-config]
+
+----
+
+NOTE: The number of vectors, the width or dimensions of the vectors and the training size can incur high CPU and memory costs as the size of each variable increases.
+This is because the training vectors have to be resident on the machine.
+
+=== Vector Index Configuration
+
+The table below displays the different configurations you can modify within your `VectorIndexConfiguration()` function.
+For more information on specific configurations, see xref:c:vector-search.adoc[Vector Search.]
+
+IMPORTANT: The beta release currently only supports the format of vectors as an array of numbers.
+
+.Vector Index Configuration Options
+[cols ="4*"]
+|===
+|Configuration Name |Is Required |Default Configuration |Further Information
+
+|Expression
+|Required
+|No default
+|A SQL++ expression indicating where to get the vectors. 
+A document property for embedded vectors
+`prediction()` to call a registered Predictive model.
+|Expression Language
+|Required
+|No default
+|Specifies the query language used. This can be SQL++ or N1Q1 query.
```

Comment:
This can be N1QL or JSON query.

For the JSON query syntax, we can link to https://github.com/couchbase/couchbase-lite-core/wiki/JSON-Query-Schema.

Reply:
Done!